app_feup/test/unit/redux/exam_action_creators_test.dart

```diff
@@ -0,0 +1,113 @@
+      final Tuple2<String, String> userPersistentInfo = Tuple2('', '');
+      final mockStore = MockStore();
+      final mockResponse = MockResponse();
+
+      final profile = Profile();
+      profile.courses = [Course(id: 7474)];
+      final content = {
+        'session': Session(authenticated: true),
+        'currUcs': [sopeCourseUnit, sdisCourseUnit],
+        'profile': profile,
+      };
+
+      when(NetworkRouter.httpClient.get(any, headers: anyNamed('headers')))
+          .thenAnswer((_) async => mockResponse);
+      when(mockResponse.statusCode).thenReturn(200);
+      when(mockStore.state).thenReturn(AppState(content));
+      test('When given a single exam', () async {
+        final Completer<Null> completer = Completer();
+        final actionCreator =
+            getUserExams(completer, parserMock, userPersistentInfo);
+        when(parserMock.parseExams(any)).thenAnswer((_) async => [sopeExam]);
+
+        actionCreator(mockStore);
+        await completer.future;
+        final List<dynamic> actions =
+            verify(mockStore.dispatch(captureAny)).captured;
+        expect(actions.length, 3);
+        expect(actions[0].status, RequestStatus.busy);
+        expect(actions[1].status, RequestStatus.successful);
+        expect(actions[2].exams, [sopeExam]);
+      });
+      test('When given two exams', () async {
+        final Completer<Null> completer = Completer();
+        final actionCreator =
+            getUserExams(completer, parserMock, userPersistentInfo);
+        when(parserMock.parseExams(any))
+            .thenAnswer((_) async => [sopeExam, sdisExam]);
+
+        actionCreator(mockStore);
+        await completer.future;
+        final List<dynamic> actions =
+            verify(mockStore.dispatch(captureAny)).captured;
+        expect(actions.length, 3);
+        expect(actions[0].status, RequestStatus.busy);
+        expect(actions[1].status, RequestStatus.successful);
+        expect(actions[2].exams, [sopeExam, sdisExam]);
+      });
+      test('When given three exams but one is to be parsed', () async {
+        final specialExam = Exam(
+            '12:00-15:00',
+            'SDIS',
+            'B119, B107, B205',
+            '2800-09-12',
+            'Exames ao abrigo de estatutos especiais - Port.Est.Especiais',
+            'Quarta');
+        final Completer<Null> completer = Completer();
+        final actionCreator =
+            getUserExams(completer, parserMock, userPersistentInfo);
+        when(parserMock.parseExams(any))
+            .thenAnswer((_) async => [sopeExam, sdisExam, specialExam]);
+
+        actionCreator(mockStore);
+        await completer.future;
+        final List<dynamic> actions =
+            verify(mockStore.dispatch(captureAny)).captured;
+        expect(actions.length, 3);
+        expect(actions[0].status, RequestStatus.busy);
+        expect(actions[1].status, RequestStatus.successful);
+        expect(actions[2].exams, [sopeExam, sdisExam]);
```

Comment:
It isn't clear why we expect only two exams to be dispatched (because of the exam type). Tests should be easier to understand. Maybe update the test's name or add a comment

Reply:
I improved the quality of the test name, hopefully that brings it out?